internal/resources/cloudprovider/data_source_azure_credential.go

```diff
@@ -0,0 +1,144 @@
+package cloudprovider
+
+import (
+	"context"
+
+	"github.com/grafana/terraform-provider-grafana/v3/internal/common"
+	"github.com/grafana/terraform-provider-grafana/v3/internal/common/cloudproviderapi"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type datasourceAzureCredential struct {
+	client *cloudproviderapi.Client
+}
+
+func makeDataSourceAzureCredential() *common.DataSource {
+	return common.NewDataSource(
+		common.CategoryCloudProvider,
+		resourceAzureCredentialTerraformName,
+		&datasourceAzureCredential{},
+	)
+}
+
+func (r *datasourceAzureCredential) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	// Configure is called multiple times (sometimes when ProviderData is not yet available), we only want to configure once
+	if req.ProviderData == nil || r.client != nil {
+		return
+	}
+
+	client, err := withClientForDataSource(req, resp)
+	if err != nil {
+		return
+	}
+
+	r.client = client
+}
+
+func (r *datasourceAzureCredential) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = resourceAzureCredentialTerraformName
+}
+
+func (r *datasourceAzureCredential) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The Terraform Resource ID. This has the format \"{{ stack_id }}:{{ resource_id }}\".",
+				Computed:    true,
+			},
+			"stack_id": schema.StringAttribute{
+				Description: "The StackID of the Grafana Cloud instance. Part of the Terraform Resource ID.",
+				Required:    true,
+			},
+			"resource_id": schema.StringAttribute{
+				Description: "The ID given by the Grafana Cloud Provider API to this Azure Credential resource.",
+				Required:    true,
+			},
+			"name": schema.StringAttribute{
+				Description: "The name of the Azure Credential.",
+				Computed:    true,
+			},
+			"client_id": schema.StringAttribute{
+				Description: "The client ID of the Azure Credential.",
+				Computed:    true,
+			},
+			"tenant_id": schema.StringAttribute{
+				Description: "The tenant ID of the Azure Credential.",
+				Computed:    true,
+			},
+			"client_secret": schema.StringAttribute{
+				Description: "The client secret of the Azure Credential.",
+				Computed:    true,
+				Sensitive:   true,
+			},
+		},
+		Blocks: map[string]schema.Block{
+			"resource_discovery_tag_filter": schema.ListNestedBlock{
```

Comment:
are these filters required to look up a unique resource? I would think that just the stack and resource ID are sufficient, because it looks like the client `GetAzureCredential` function only needs the stack ID and resource ID

Reply:
if so, then it looks like `resource_discovery_tag_filter` needs to be retrieved and set below from the API response as well

Reply:
Done